src/LndGrpc/Client.hs

```diff
@@ -32,11 +34,14 @@ runUnary ::
   p (Either LndError res)
 runUnary rpc env req = do
   res <-
-    liftIO (Right <$> this)
-      `catch` (\(e :: BlockedIndefinitelyOnMVar) -> pure $ Left e)
+    catchExceptions $ liftIO (Right <$> this)
   pure $ case res of
     Right (Right (Right (Right (_, _, Right x)))) ->
       Right x
+    Right (Right (Right (Right (_, _, Left "wallet not created, create one to enable full RPC access")))) ->
+      Left $ LndLockedError "wallet not created"
+    Right (Right (Right (Right (_, _, Left "wallet locked, unlock it to enable full RPC access")))) ->
+      Left $ LndLockedError "wallet locked"
```

Comment:
Let's have 2 separate constructors for Locked (non fatal error) and NonExistent (fatal) wallet errors.

Reply:
Done